Adds task transform application

cppwfs opened the pull request:
resolves #203

cppwfs committed:
Adds task transform application

resolves #203

artembilan commented on processor/spring-cloud-starter-stream-processor-tasklaunchrequest-transform/src/main/java/org/springframework/cloud/stream/app/tasklaunchrequest/transform/processor/TasklaunchrequestTransformProcessorConfiguration.java:
```diff
@@ -0,0 +1,79 @@
+				processorProperties.getUri(), commandLineArgs, properties,
+				deploymentProperties);
+
+		return new GenericMessage<TaskLaunchRequest>(request);
```
I'm not sure if there is some environment in SCSt which should be supplied with some specific header, but I don't see reason here to instantiate `Message` ourselves. More over we don't provide any particular header.
OTOH and it is my main concern here: if `transformer` returns a `Message<?>` it doesn't populate any additional headers.
So, just returning `request` will make a day for you!
The same `Message<?>` as a further logic you have. Plus we carry headers if that.

cppwfs replied:
Sounds good to me.  Code will be updated to return the request.

artembilan commented on processor/spring-cloud-starter-stream-processor-tasklaunchrequest-transform/src/main/java/org/springframework/cloud/stream/app/tasklaunchrequest/transform/processor/TasklaunchrequestTransformProcessorProperties.java:
```diff
@@ -0,0 +1,119 @@
+	private String commandLineArguments;
+
+	/**
+	 * Comma delimited list of deployment properties to be applied to the
```
"Space delimited", " Comma delimited"...

Do you have any plans for any other delimiter styles in other options? :wink: 

I mean what is the point of the mix if we are here in the options for one application?
Thanks

cppwfs replied:
This is to remain consistent in how a user will utilize tasks.  i.e. command line parameters are  space delimited while deployment properties are comma delimited.   We could choose just one, but in this case I chose to be consistent.

artembilan commented on processor/spring-cloud-starter-stream-processor-tasklaunchrequest-transform/src/main/java/org/springframework/cloud/stream/app/tasklaunchrequest/transform/processor/ParsingUtils.java:
```diff
@@ -0,0 +1,136 @@
+import org.springframework.util.StringUtils;
+
+/**
+ * Provides utility methods for formatting and parsing deployment properties.
```
Looks like this class parses not only "deployment properties", but also "commandLineArgs".
In addition it is intended for public usage. Plus its name is so generic. 
Maybe `TaskLaunchRequestParser`?..
Although this class can be fully moved as a bunch of private methods to the `TasklaunchrequestTransformProcessorConfiguration`

cppwfs replied:
Will move these methods into the config as private methods.

artembilan commented on processor/spring-cloud-starter-stream-processor-tasklaunchrequest-transform/src/main/java/org/springframework/cloud/stream/app/tasklaunchrequest/transform/processor/TasklaunchrequestTransformProcessorConfiguration.java:
```diff
@@ -0,0 +1,79 @@
+import org.springframework.util.StringUtils;
+
+/**
+ * A transform that takes the maven repository coordinates, command line
```
A transform**er**

artembilan commented on processor/spring-cloud-starter-stream-processor-tasklaunchrequest-transform/src/main/java/org/springframework/cloud/stream/app/tasklaunchrequest/transform/processor/TasklaunchrequestTransformProcessorConfiguration.java:
```diff
@@ -0,0 +1,79 @@
+		Map<String, String> deploymentProperties = null;
+		List<String> commandLineArgs = null;
+
+		if(StringUtils.hasText(processorProperties.getDataSourceUrl())){
```
Code style nit-pick: whitespace before `{`.
There are many other similar code style issues.
But, eh, that is minor and can be fixed with simple reformat during merge.

cppwfs replied:
oops :-)

cppwfs committed:
Updated based on code review
A little more cleanup

mminella commented on processor/spring-cloud-starter-stream-processor-tasklaunchrequest-transform/src/main/java/org/springframework/cloud/stream/app/tasklaunchrequest/transform/processor/TasklaunchrequestTransformProcessorConfiguration.java:
```diff
@@ -0,0 +1,180 @@
+			commandLineArgs = parseParams(processorProperties.getCommandLineArguments());
+		}
+
+		properties.put("payload", message);
```
Why is this here?

cppwfs replied:
we can drop it.  But thought it could be used by a task downstream.

cppwfs replied:
Payload no longer added to properties.

cppwfs committed:
Updated the parsing string for deployment  params

ilayaperumalg commented on processor/spring-cloud-starter-stream-processor-tasklaunchrequest-transform/README.adoc:
```diff
@@ -0,0 +1,28 @@
+
+== Options
+
+The **$$filter$$** $$processor$$ has the following options:
```
This should be `tasklaunchrequest`

cppwfs replied:
updated

cppwfs committed:
Per Code Review:

Payload no longer added to environment variables of TaskLaunchRequest
Docs updated.

ilayaperumalg commented:
LGTM, merging.

ilayaperumalg commented:
Squashed the commits, rebased and merged as  ca2dc8f
